src/internal/kopia/wrapper.go

```diff
@@ -497,22 +505,48 @@ func (w Wrapper) BackupCollections(
 		model.ServiceTag: service.String(),
 	}
 
-	dirTree, err := inflateDirTree(ctx, collections, progress)
+	dirTree, serviceCats, resourceOwners, err := inflateDirTree(ctx, collections, progress)
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "building kopia directories")
 	}
 
-	s, err := w.makeSnapshotWithRoot(ctx, dirTree, progress)
+	s, err := w.makeSnapshotWithRoot(ctx, dirTree, serviceCats, resourceOwners, progress)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	return s, progress.deets, nil
 }
 
+func serviceCatTag(s path.ServiceType, c path.CategoryType) string {
```

Comment:
api here seems unnecessarily one step distanced.  Any reason to to take `serviceCatTag(path)`?  That would be consistent with all of your invocations.